zebra-chain/src/sapling/tree.rs

```diff
@@ -13,33 +13,51 @@
 #![allow(clippy::unit_arg)]
 #![allow(dead_code)]
 
-use std::{fmt, io};
+use std::{collections::VecDeque, fmt};
 
 use bitvec::prelude::*;
+use lazy_static::lazy_static;
 #[cfg(any(test, feature = "proptest-impl"))]
 use proptest_derive::Arbitrary;
 
-use crate::serialization::{SerializationError, ZcashDeserialize, ZcashSerialize};
+use super::commitment::{pedersen_hashes::pedersen_hash, NoteCommitment};
 
-use super::commitment::pedersen_hashes::pedersen_hash;
+const MERKLE_DEPTH: usize = 32;
 
 /// MerkleCRH^Sapling Hash Function
 ///
-/// MerkleCRH^Sapling(layer, left, right) := PedersenHash(“Zcash_PH”, l || left ||right)
-/// where l = I2LEBSP_6(MerkleDepth^Sapling − 1 − layer)
+/// MerkleCRH^Sapling(layer, left, right) := PedersenHash(“Zcash_PH”, l || left || right)
+/// where l = I2LEBSP_6(MerkleDepth^Sapling − 1 − layer) and
+/// left, right, and the output are all technically 255 bits (l_MerkleSapling), not 256.
 ///
 /// https://zips.z.cash/protocol/protocol.pdf#merklecrh
```

Comment:
This doc comment seems very spec oriented, might be a good idea to also add some more layman oriented explanations of what this function does and when it should be used?

Reply:
I'm adding a little bit but it should only be used internally to computing these note commitment merkle trees